src/Shared/FileMatcher.cs

```diff
@@ -2275,6 +2296,53 @@ private static int SkipSlashes(string aString, int startingIndex)
             return index;
         }
 
+        /// <summary>
+        /// Returns true if drive enumerating wildcard patterns are detected using the directory and wildcard parts.
+        /// </summary>
+        private static bool IsDriveEnumeratingWildcardPattern(string directoryPart, string wildcardPart)
+        {
+            int directoryPartLength = directoryPart.Length;
+            int wildcardPartLength = wildcardPart.Length;
+            if (directoryPartLength >= 3 &&                                                                  // Handles <drive letter>:<slashes>** cases
+                wildcardPartLength >= 2 &&
+                IsDrivePatternWithoutSlash(directoryPart[0], directoryPart[1]))
+            {
+                for (int i = 2; i < directoryPartLength-1; i++)
+                {
+                    if (!FileUtilities.IsAnySlash(directoryPart[i]))
+                    {
+                        return false;
+                    }
+                }
+
+                return IsFullFileSystemScan(directoryPart[directoryPartLength - 1], wildcardPart[0], wildcardPart[1]);
+            }
+            else if (directoryPartLength == 1 &&                                                             // Handles \** case
```

Comment:
Is there a `////////**` case?

Reply:
Added a `c:////////**` case, but `////////**` does not result in drive enumeration (even when I manually test using current msbuild bits locally on my Windows m/c), unless I'm misunderstanding this case.